### ISSUE TITLE
A `@type_check_only` warning when defining a type symbol for use in both type checking and runtime by leveraging `TYPE_CHECKING`.

### ISSUE THREAD

**Issue by junkmd** [microsoft/pyright#9934]
## Actual Behavior:
A `@type_check_only` warning is raised when a type symbol with the same name is alias-imported in the `TYPE_CHECKING` block and alias-assigned in the `else` block.

This issue occurs in cases where the metaclass of `ctypes.Structure` is directly used.

```py
from ctypes import Structure
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from _ctypes import _PyCStructType as meta
else:
    meta = type(Structure)

class Sub(Structure):
    ...

typ = meta("name", (Sub,), {})
#     ^^^^ "_PyCStructType" is marked as @type_check_only and can be used only in type annotations

```

I understand that `_PyCStructType` is marked with the `@type_check_only` decorator in typeshed. My point is that when defining a type symbol for use in both type checking and runtime by leveraging `TYPE_CHECKING`, it would be preferable if this warning were not displayed.

## Expected Behavior:

The `@type_check_only` warning is not displayed.

## Environment:

* Pylance version: 1.1.394
* Python version: 3.13

**Comment by erictraut** [comment]
Pyright is working as intended here, so this isn't a bug.

The `TYPE_CHECKING` symbol is effectively a way to lie to a type checker about what is going to be seen at runtime. Type checkers don't treat symbols differently if they are imported under an `if TYPE_CHECKING` check. You'll see the same behavior in other Python type checkers.
